File: personal-assistant-agent/src/agents/calendarAgent.ts
import { BaseAgent } from './base';

export class CalendarAgent extends BaseAgent {
    async fetchDailyMeetingSchedule(date: string) {
        const { fetchDailyMeetingSchedule } = await import('../tools/FetchDailyMeetingSchedule');
        return fetchDailyMeetingSchedule(this.google.oauth2Client, date);
    }
}

File: personal-assistant-agent/src/agents/calendar_agent.py
from src.agents.base import BaseAgent
from src.tools.fetch_daily_meeting_schedule import (
    DailyScheduleResult,
    fetch_daily_meeting_schedule,
)


class CalendarAgent(BaseAgent):
    def fetch_daily_meeting_schedule(self, date: str) -> DailyScheduleResult:
        if not self.google.oauth2_client:
            raise RuntimeError("Google client not authenticated")
        return fetch_daily_meeting_schedule(self.google.oauth2_client, date)
